pkg/cloudprovider/aws/packing/packing.go

```diff
@@ -77,33 +90,76 @@ func (p *podPacker) Pack(ctx context.Context, constraints *cloudprovider.Constra
 	return packings, nil
 }
 
-// TODO filter instance types based on node contraints like availability zones etc.
-func (p *podPacker) getNodeCapacities(constraints *cloudprovider.Constraints) []*nodeCapacity {
+func (p *podPacker) getNodeCapacities(constraints *cloudprovider.Constraints) ([]*nodeCapacity, error) {
 	result := make([]*nodeCapacity, 0)
-	for _, nc := range nodeCapacities {
-		ncc := nc.Copy()
-		kubeletOverhead := binpacking.CalculateKubeletOverhead(ncc.total)
-		if ok := ncc.reserve(resources.Merge(constraints.Overhead, kubeletOverhead)); !ok {
-			zap.S().Errorf("Failed to reserve kubelet overhead for node capacity type %v", ncc.instanceType)
+
+	describeInstanceTypesInput := &ec2.DescribeInstanceTypesInput{
+		Filters: constraintsToDescribeInstanceTypesFilters(constraints),
+	}
+
+	err := p.ec2.DescribeInstanceTypesPagesWithContext(context.TODO(), describeInstanceTypesInput, func(page *ec2.DescribeInstanceTypesOutput, lastPage bool) bool {
+		for _, instanceTypeInfo := range page.InstanceTypes {
+			nc := instanceTypeInfoToNodeCapacity(*instanceTypeInfo)
+			kubeletOverhead := binpacking.CalculateKubeletOverhead(nc.total)
+			if ok := nc.reserve(resources.Merge(constraints.Overhead, kubeletOverhead)); !ok {
+				zap.S().Errorf("Failed to reserve kubelet overhead for node capacity type %v", nc.instanceType)
```

Comment:
Do we want to error here? Is debug better? IIUC, this only errors if the theoretical node is full, which might happen multiple times in simulation.

Reply:
I'd probably not consider it an error even - it means the node cannot handle kubelet overhead at all, so we shouldn't consider it - maybe make this a log info ("excluding ...")?

Reply:
I think that makes sense for info level. I just brought this in from a rebase, so not sure if there was a reason for it being error.

Reply:
We're really judicious about Info level -- we can talk about the ideology here, maybe debug?

Reply:
I don't have a super strong opinion, so I can go either way.  I think Jacob is right that it shouldn't ever happen the way the kubelet overhead is being calculated right now since it's a scaled a percentage. But I suppose if there was a weird instance type that was released, it could spam everyone using karpenter if it's at info level, so I'm fine with debug.